src/dashboard/src/components/administration/forms.py

```diff
@@ -129,10 +139,306 @@ class AtomDipUploadSettingsForm(SettingsForm):
+            'type': 'replace_dict',
+            'name': 'compression_level',
+            'label': 'Select compression level',
+        },
+
+        #
+        # Storage service locations
+        # Required property: purpose
+        #
+        'b7a83da6-ed5a-47f7-a643-1e9f9f46e364': {
+            'type': 'storage_service',
+            'name': 'store_dip_location',
+            'label': 'Store DIP location',
+            'purpose': 'DS',
+        },
+        'b320ce81-9982-408a-9502-097d0daa48fa': {
+            'type': 'storage_service',
+            'name': 'store_aip_location',
+            'label': 'Store AIP location',
+            'purpose': 'AS',
+        },
+
+        #
+        # Time fields (days)
+        # Optional property: placeholder, min_value
+        # Required properties: chain (UUID)
+        #
+        '19adb668-b19a-4fcb-8938-f49d7485eaf3': {
+            'type': 'days',
+            'name': 'quarantine_expiry_days',
+            'label': 'Remove from quarantine after (days)',
+            'placeholder': 'days',
+            'chain': '333643b7-122a-4019-8bef-996443f3ecc5',
+            'min_value': 0,
+        }
+    }
+
+    EMPTY_OPTION_NAME = _('None')
+    EMPTY_CHOICES = [
+        (None, EMPTY_OPTION_NAME),
+    ]
+    DEFAULT_FIELD_OPTS = {
+        'required': False,
+        'initial': None,
+    }
+
+    name = forms.RegexField(max_length=16, regex=r'^\w+$', required=True)
+
+    def __init__(self, *args, **kwargs):
+        super(ProcessingConfigurationForm, self).__init__(*args, **kwargs)
+        for choice_uuid, field in self.processing_fields.iteritems():
+            ftype = field['type']
+            opts = self.DEFAULT_FIELD_OPTS.copy()
+            if 'label' in field:
+                opts['label'] = field['label']
+            if ftype == 'days':
+                if 'min_value' in field:
+                    opts['min_value'] = field['min_value']
+                self.fields[choice_uuid] = forms.IntegerField(**opts)
+                if 'placeholder' in field:
+                    self.fields[choice_uuid].widget.attrs['placeholder'] = field['placeholder']
+            else:
+                choices = opts['choices'] = list(self.EMPTY_CHOICES)
+                if ftype == 'boolean':
+                    if 'yes_option' in field:
+                        choices.append((field['yes_option'], 'Yes'))
+                    if 'no_option' in field:
+                        choices.append((field['no_option'], 'No'))
+                elif ftype == 'chain_choice':
+                    link = models.MicroServiceChainLink.objects.get(pk=choice_uuid)
+                    chain_choices = models.MicroServiceChainChoice.objects.filter(choiceavailableatlink_id=link.pk)
+                    ignored_choices = field.get('ignored_choices')
+                    for item in chain_choices:
+                        chain = item.chainavailable
+                        if ignored_choices is not None:
```

Comment:
Tip: If above is `ignored_choices = field.get('ignored_choices', [])` then you don't need to check 'if ignore_choices is None'.  Second parameter `.get()` is default value.